metrics/network/remote_network/remote-networking-test-common.sh

```diff
@@ -0,0 +1,120 @@
+#!/bin/bash
+
+# Copyright (c) 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# This file contains functions that are shared among the networking
+# tests that are using iperf
+
+SCRIPT_PATH=$(dirname "$(readlink -f "$0")")
+source "${SCRIPT_PATH}/../../lib/common.bash"
+
+# Argument of the test to run
+argument="$1"
+# Name of the interface were swarm will run
+interface_name="$2"
+# Name of the user (ssh)
+ssh_user="$3"
+# Ip address (ssh)
+ssh_address="$4"
+# Image for network testing
+network_image="gabyct/network"
+# Number of replicas that will be launch in each host
+number_of_replicas=1
+# Timeout in seconds to verify replicas are running
+timeout=10
+# Name of the first service
+first_service="testswarm1"
+# Name of the second service
+second_service="testswarm2"
+# Port where the first service will run
+port_first_service="8080:80"
+# Port where the second service will run
+port_second_service="8081:80"
+
+# This function will start swarm as well as it will label
+# the nodes and create the replicas
+function setup_swarm {
+	token=$($DOCKER_EXE swarm init --advertise-addr "$interface_name")
+	token_name=$(echo "$token" | grep "token" | head -1 | cut -d '\' -f1)
+	ip_name=$(echo "$token" | grep ":" | tail -1)
+	ssh "$ssh_user"@"$ssh_address" "$DOCKER_EXE swarm join $token_name $ip_name"
+	first_node=$($DOCKER_EXE node ls -q | head -1)
+	second_node=$($DOCKER_EXE node ls -q | tail -1)
+	$DOCKER_EXE node update --label-add machine=machine1 "$first_node"
+	$DOCKER_EXE node update --label-add machine=machine2 "$second_node"
+	$DOCKER_EXE service create \
+		--name "$first_service" --replicas "$number_of_replicas" \
+		--constraint 'node.labels.machine == machine1' \
+		--publish "$port_first_service" "$network_image" sh
+	$DOCKER_EXE service create \
+		--name "second_service" --replicas "$number_of_replicas" \
+		--constraint 'node.labels.machine == machine2' \
+		--publish "$port_second_service" "$network_image" sh
+}
+
+# This function will verify that the replica on the host client is running
+function client_replica_status {
+	for i in $(seq "$timeout"); do
+		client_status=$($DOCKER_EXE ps -q --filter=status=running | wc -l)
+		if [ "$client_status" -ge "$number_of_replicas" ]; then
+			break
+		fi
+		sleep 1
+	done
+}
+
+# This function will verify that the replica on the host server is running
+function server_replica_status {
+	status=$(mktemp)
```

Comment:
maybe could be possible to use `$()` for getting the output instead to use tmp files then you could avoid to use `rm` lines